README.md

```diff
@@ -26,35 +26,120 @@ flutter pub add flame_behaviors
 
 ## Usage ✨
 
-### Behavior
+### Entity
+
+The entity is the building block of a game. It represents a visual game object that can hold 
+multiple `Behavior`s which in turn define how the entity behaves.
+
+```dart
+// Define a custom entity by extending `Entity`.
+class MyEntity extends Entity {
+  MyEntity() : super(behaviors: [MyBehavior()]);
+}
+```
 
-A behavior is a component that defines how an entity behaves. It can be attached to an `Entity` and handle a specific behavior for that entity.
+### Behavior
 
-A behavior can have it's own `Component`s for adding extra functionality related to the behavior. It cannot, however, have its own `Behavior`s.
+A behavior is a component that defines how an entity behaves. It can be attached to an `Entity` 
+and handle a specific behavior for that entity. Behaviors can either be generic for any entity 
+or you can specify the specific type of entity it requires:
 
 ```dart
-// Define a custom behavior by extending `Behavior`.
-class CollisionBehavior extends Behavior with CollisionCallbacks {
-  CollisionBehavior(this._hitbox) : super(children: [_hitbox]);
+// Can be added to any type of Entity.
+class MyGenericBehavior extends Behavior {
+  ...
+}
+
+// Can only be added to MyEntity and subclasses of it.
+class MySpecificBehavior extends Behavior<MyEntity> {
+  ...
+}
+```
 
-  final RectangleHitbox _hitbox;
+Each behavior can have it's own `Component`s for adding extra functionality related to the 
+behavior. For instance a `TimerComponent` to have time based behavioral activity:
 
+```dart
+class MyBehavior extends Behavior {
   @override
-  Future<void> onLoad() async {...}
+  Future<void> onLoad() async {
+    await add(TimerComponent(period: 5, repeat: true, onTick: _onTick));
+  }
+
+  void _onTick() {
+    // Do something every 5 seconds.
+  }
+}
+```
 
+**Note**: A behavior, by definition, cannot have it's own `Behavior`s. 
```

Comment:
```suggestion
**Note**: A behavior, by definition, cannot have its own `Behavior`s. 
```
Why not? Can we add another sentence as to why? Was it just an implementation choice? It's not super obvious to me on first read.